src/backend/src/controllers/change-requests.controllers.ts

```diff
@@ -76,8 +71,101 @@ export const reviewChangeRequest = async (req: Request, res: Response) => {
         approved: true
       }
     });
-  }
+    const wbs = await prisma.wBS_Element.findUnique({
+      where: {
+        wbsElementId: foundCR.wbsElementId
+      },
+      include: {
+        workPackage: true,
+        project: true
+      }
+    });
+
+    if (!wbs) {
+      return res.status(404).json({ message: `wbs element with id #${foundCR.wbsElementId} not found` });
+    }
+
+    const wp = { ...wbs.workPackage };
+
+    const proj = { ...wbs.project };
+
+    if (!wp && proj) {
+      let updatedBudgetImpact;
+      if (!proj.budget) {
+        updatedBudgetImpact = 0 + foundPs.budgetImpact;
+      } else {
+        updatedBudgetImpact = proj.budget + foundPs.budgetImpact;
+      }
+      const change = {
+        changeRequestId: crId,
+        implementerId: reviewerId,
+        detail: buildChangeDetail('Budget Impact', String(proj.budget), String(updatedBudgetImpact))
+      };
+      await prisma.project.update({
+        where: { projectId: wbs.projectNumber },
+        data: {
+          budget: updatedBudgetImpact,
+          wbsElement: {
+            update: {
+              changes: {
+                createMany: {
+                  data: [change]
+                }
+              }
+            }
+          }
+        }
+      });
+    } else if (wp) {
+      const wpProj = await prisma.project.findUnique({
+        where: { projectId: wp.projectId }
+      });
+      if (!wpProj) {
+        return res.status(400).json({ message: 'Work package project not found' });
+      }
+      const updatedBudgetImpact = wpProj.budget + foundPs.budgetImpact;
+      let updatedDuration;
+      if (!wp.duration) {
+        updatedDuration = 0 + foundPs.timelineImpact;
+      } else {
+        updatedDuration = wp.duration + foundPs.timelineImpact;
+      }
 
+      const changes = [
+        {
+          changeRequestId: crId,
+          implementerId: reviewerId,
+          detail: buildChangeDetail('Budget Impact', String(wpProj.budget), String(updatedBudgetImpact))
+        },
+        {
+          changeRequestId: crId,
+          implementerId: reviewerId,
+          detail: buildChangeDetail('Timeline Impact', String(wp.duration), String(updatedDuration))
+        }
+      ];
+
+      await prisma.work_Package.update({
+        where: { workPackageId: wbs.workPackageNumber },
+        data: {
+          project: {
+            update: {
+              budget: updatedBudgetImpact
+            }
+          },
+          duration: updatedDuration,
+          wbsElement: {
+            update: {
+              changes: {
+                createMany: {
+                  data: changes
```

Comment:
im not sure but i wonder if the project budget change should be registered on the project and not the work package

Reply:
Do you mean instead of awaiting a work_package with a project, I should await a project with a project and do the changes there?

Reply:
im saying a project has a list of changes and a work package has a list of changes so which one should we put it on if we're changing the project's data but it's because of a proposed solution to the work package? i dont really know the answer tbh